File: frontend/src/lib/types/employee.ts
/**
 * Catalyst API — Employee types.
 *
 * Mirrors the FastAPI backend's Employee Pydantic models.
 */

export interface Employee {
  id: string;
  name: string;
  department: string;
  role: string;
  email: string;
  phone: string;
}

export type EmployeeCreate = Omit<Employee, "id">;

File: backend/app/schemas/employee.py
from typing import Optional
from pydantic import BaseModel, Field


class EmployeeBase(BaseModel):
    name: str = Field(..., description="Full name of the employee")
    department: str = Field(..., description="Department name")
    role: str = Field(..., description="Job role or title")
    email: str = Field(..., description="Email address")
    phone: str = Field(..., description="Phone number")


class EmployeeCreate(EmployeeBase):
    pass


class EmployeeUpdate(BaseModel):
    name: Optional[str] = None
    department: Optional[str] = None
    role: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None


class Employee(EmployeeBase):
    id: str = Field(..., description="Unique identifier for the employee")

    class Config:
        from_attributes = True
